### PR TITLE
Make documentation great again

### PR DESCRIPTION
**Pull request by colthreepv**
Joke name aside, this project really needs friendlier documentation (see #31  #33)

I'm putting a revamped README for 2.x release, pinging any recent contributor to have some feedback in form of comments (or PR if you have time)

@AndrewKeig @iheanyi @gdw2 @aymericbeaumet @NikolaBorislavovHristov @holm  @gedennis @unixdev 

I'm not a native english speaker, feel free to correct phrasing or suggest less convoluted ways to infuse knowledge

### CODE REVIEW

**Comment by isaachinman** [comment]
@colthreepv There seems to be a lot in this PR that is not documentation. Mind explaining?

**Comment by colthreepv** [comment]
Sorry, you are completely right.  
The `documentation` branch departs from the branch `2.x` (those branches are also present in this current repository)

I should change the target for this PR, but does not seem possible to edit this detail.

**Comment by isaachinman** [comment]
Do you mean you want the base of this PR to be the `2.x` branch?

**Comment by colthreepv** [comment]
Exactly

**Comment by colthreepv** [comment]
it was possible to do in the interface, just a little bit hidden. Done

File: README.md [hunk 1]
```diff
@@ -118,41 +134,21 @@ Full code for these examples is to be found in [`test/`](test/) directory.
 
 ## How to use
 
-### `req` objects gets parsed
-When `Joi` validates the `body`, `params`, `query`, `headers` or `cookies` it returns it as Javascript Object.
+### Request properties and parsing
+`express-validation` relies on express.js to parse request bodies.  
+It's the developer duty to correctly register json parser(s), multipart parser (s), cookie parser(s) and so on.
```
Comment (aymericbeaumet) [inline README.md]: `parser(s)`

File: README.md [hunk 2]
```diff
@@ -1,49 +1,65 @@
...
-[![build status](https://travis-ci.org/AndrewKeig/express-validation.svg)](http://travis-ci.org/AndrewKeig/express-validation)
-
 ## Install
-
 ```sh
 $ npm install express-validation --save
 ```
 
+## Difference with express-validator
+There is a more star(red) and famous express.js middleware that covers the same use case as this one, they do not overlap at all, as they offer different approaches.
```
Comment (isaachinman) [inline README.md]: "_but_ they do not overlap at all"

File: README.md [hunk 3]
```diff
@@ -1,49 +1,65 @@
...
+
+Validat**or** offers an imperative syntax - meaning every check has to be explicitly declared as code, inside a function, middleware in this case, and while it might be comfortable and easier to read the first couple times, it can get clunky in the long run.
```
Comment (isaachinman) [inline README.md]: I'd put "middleware in this case" inside parentheses and use less commas.

File: README.md [hunk 4]
```diff
@@ -1,49 +1,65 @@
...
 
-## Supporting
+Validat**ion** it's strongly based on [Joi][joi], and leverages a declarative syntax. This means developers declare how to handle validation errors, and using Joi syntax to shape the request payloads. API errors tend to be uniform as they are all generated by Joi and handled in your error function.
```
Comment (isaachinman) [inline README.md]: "_is_ strong based"

"and _use_ Joi syntax"

File: README.md [hunk 5]
```diff
@@ -289,17 +284,55 @@ ev.options();
 ```
 Thanks to node `require()` caching, all the other `express-validation` instances also have the same set of global options.
 
+### Request mutation
+An choice for advanced users can be to enable or disable request mutation.
```
Comment (isaachinman) [inline README.md]: "_A_ choice"

**Comment by brajchit** [comment]
In the v2.0 with Joi v.15, the `field` in `errors` response is an Array of strings instead of `string`.

**Comment by AndrewKeig** [comment]
Closing, released 3.0

**Pull request closed without merging**